Vagrantfile

```diff
@@ -0,0 +1,100 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Read the Kata Containers CI job from the CI_JOB environment variable.
+job = ENV['CI_JOB'] || ""
+guest_home_dir = '/home/vagrant'
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+
+  # By default vagrant sync the current directory. Let's disabled it because the directory
+  # will be synced later to the proper destination.
+  config.vm.synced_folder ".", "/vagrant", disabled:true
+  # TODO: Ideally we should be using nfs or 9p for bidirectional syncing
+  config.vm.synced_folder ".", "#{guest_home_dir}/go/src/github.com/kata-containers/tests", type:"rsync"
+  config.vm.synced_folder "#{ENV['GOPATH']}/src/github.com/kata-containers/kata-containers",
+    "#{guest_home_dir}/go/src/github.com/kata-containers/kata-containers", type:"rsync"
+
+  config.vm.provider "libvirt" do |lv|
+    lv.driver = "kvm"
+    lv.cpus = "4"
+    lv.memory = "8192"
+  end
+
+  # Shared provision script.
+  config.vm.provision "shell", env: {"CI_JOB" => job}, inline: <<-SHELL
+
+    export GOPATH="#{guest_home_dir}/go"
+    kata_tests_repo_dir="${GOPATH}/src/github.com/kata-containers/tests"
+
+    env_file="#{guest_home_dir}/ci_job_env"
+    cat <<-EOF > ${env_file}
+export GOPATH="$GOPATH"
+export PATH="/usr/local/go/bin:\$GOPATH/bin:\$PATH"
+export CI="true"
+export CI_JOB="$CI_JOB"
+pushd $kata_tests_repo_dir
+source .ci/ci_job_flags.sh
+popd
+EOF
+
+    source ${env_file}
+
+    # Customize the .bashrc so that it will have the variables exported
+    # after log-in.
+    echo "cd $kata_tests_repo_dir" >> #{guest_home_dir}/.bashrc
+    echo "source $env_file" >> #{guest_home_dir}/.bashrc
+
+    distro=$(source /etc/os-release; echo $ID)
+    case "$distro" in
+      "fedora")
+        # TODO: The kubernetes tests. Maybe it should be handled on
+        # integration/kubernetes/init.sh
+        sudo systemctl stop firewalld || true
+        sudo swapoff -a || true
+
+        # Fedora >= 32 Kernel comes configured with cgroup v2 by default.
+        # This switches back to cgroup v1. It requires a reboot.
+        sudo dnf install -y grubby
+        sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+
+        # It tries to build a ubuntu based image by default. When it pulls the ubuntu image
+        # from docker.io it can hit the pulling ratio error. Let's use fedora.
```

Comment:
Can we pull from `quay.io`/AWS/blah instead?

Reply:
I'm not sure about others but here osbuilder fails often due that docker.io limitation. I'd advocate to pull the image from aws storage as you suggest. However I prefer to not touch on it on this PR.